File: backend/Ledgerly.Application/DTOs/Dashboard/OutstandingByCustomerDto.cs
namespace Ledgerly.Application.DTOs.Dashboard;

public class OutstandingByCustomerDto
{
    public Guid CustomerId { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public decimal TotalOutstanding { get; set; }
    public DateTime? LastBillDate { get; set; }
    public int OpenBillsCount { get; set; }
    public List<CustomerBillDto> Bills { get; set; } = new();
}

public class CustomerBillDto
{
    public Guid BillId { get; set; }
    public string? BillNumber { get; set; }
    public decimal OutstandingAmount { get; set; }
    public DateTime BillDate { get; set; }
}

File: app/lib/models/dashboard/outstanding_by_customer_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'customer_bill_dto.dart';

part 'outstanding_by_customer_dto.g.dart';

@JsonSerializable()
class OutstandingByCustomerDto {
  final String customerId;
  final String customerName;
  final double totalOutstanding;
  @JsonKey(name: 'lastBillDate')
  final DateTime? lastBillDate;
  final int openBillsCount;
  final List<CustomerBillDto> bills;

  OutstandingByCustomerDto({
    required this.customerId,
    required this.customerName,
    required this.totalOutstanding,
    this.lastBillDate,
    required this.openBillsCount,
    required this.bills,
  });

  factory OutstandingByCustomerDto.fromJson(Map<String, dynamic> json) =>
      _$OutstandingByCustomerDtoFromJson(json);

  Map<String, dynamic> toJson() => _$OutstandingByCustomerDtoToJson(this);
}

